### PR TITLE
feat(#41): latex.rs, puzzles

### LINKED ISSUE #41: Generate report.tex with all interaction history with fakehub
Let's generate `report.tex` file that will contain all the history of interaction with our tool during the particular run session. We can even render a nice PDF document from that tex file if user pass `--report` flag within `start` command.

### PR DESCRIPTION
**Pull request by h1alexbel**
@l3r8yJ take a look, please
closes #41

---

## PR-Codex overview
This PR adds a new `report` module with `latex.rs` for generating reports in LaTeX format. It also includes a function to read a LaTeX template file and a test for template content comparison.

### Detailed summary
- Added `report` module with `latex.rs` for report generation
- Implemented function to read LaTeX template file
- Added a test for comparing template content

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: server/src/report/latex.rs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+/// # Examples
+///
+/// ```
+/// use crate::server::report::latex::template;
+/// let content = template(None);
+/// print!("{content}")
+/// ```
+pub fn template(path: Option<&str>) -> String {
+    return fs::read_to_string(Path::new(path.unwrap_or("resources/report.tex")))
+        .unwrap();
```
Comment (l3r8yJ) [inline server/src/report/latex.rs:44]: @h1alexbel it's not the best practice to use `#unwrap`, my suggestion is to setup `clippy` to exclude usage of `#unwrap`

**Review by l3r8yJ — CHANGES_REQUESTED** [review]
@h1alexbel take a look at my comments below, please. Also I'm not sure that this PR is closing the tickets, should be use `ref` in description instead of `closing` or refactor PR to fully close the issue, wdyt?

**Comment by h1alexbel** [comment]
> @h1alexbel take a look at my comments below, please. Also I'm not sure that this PR is closing the tickets, should be use `ref` in description instead of `closing` or refactor PR to fully close the issue, wdyt?

Let me explain then. #41 is quite abstract and we had a conversation in it about additional flags and so on. I decided to "kick off" this ticket here, and decompose future work. That's why I committed a few related puzzles as well. WDYT?

File: server/src/report/latex.rs (hunk 1 above)
Comment (h1alexbel) [reply server/src/report/latex.rs:44]: @l3r8yJ Should I use `match` instead?

**Comment by h1alexbel** [comment]
need to adjust puzzles according to this comment: https://github.com/h1alexbel/fakehub/issues/41#issuecomment-2186609346

**Comment by l3r8yJ** [comment]
@h1alexbel can you refactor puzzles and request again?

**Comment by h1alexbel** [comment]
@l3r8yJ yes

**Comment by h1alexbel** [comment]
@l3r8yJ the branch is ready

File: server/src/report/latex.rs [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+/// returns: String
+///
+/// # Examples
+///
+/// ```
+/// use crate::server::report::latex::template;
+/// let content = template(None);
+/// print!("{content}")
+/// ```
+pub fn template(path: Option<&str>) -> String {
```
Comment (l3r8yJ) [inline server/src/report/latex.rs:42]: @h1alexbel i think in this case path should be required, you can add it to `Args` and set default value if nothing was provided to `--report` argument; `txt` would be fine for default value

File: server/src/report/latex.rs [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+mod tests {
+    use anyhow::Result;
+    use std::fs::File;
+    use std::io::Write;
+    use tempdir::TempDir;
+
+    use crate::report::latex::template;
+
+    #[test]
+    fn returns_default_template() -> Result<()> {
```
Comment (l3r8yJ) [inline server/src/report/latex.rs:57]: @h1alexbel let's add a puzzle to use something like `@ExtendsWith`  to pass `expected` as parameter. If such library doesn't exists we can create one (also can be added to puzzle description)

**Review by l3r8yJ — CHANGES_REQUESTED** [review]
@h1alexbel take a look at my comments below, please

File: server/src/report/latex.rs (hunk 2 above)
Comment (h1alexbel) [reply server/src/report/latex.rs:42]: @l3r8yJ this function will be called only if cli parser will find non-empty `--report` option

File: server/src/report/latex.rs (hunk 1 above)
Comment (l3r8yJ) [reply server/src/report/latex.rs]: @h1alexbel sure

File: server/src/report/latex.rs (hunk 2 above)
Comment (l3r8yJ) [reply server/src/report/latex.rs]: @h1alexbel can you add the puzzle or implementation?

Comment (h1alexbel) [reply server/src/report/latex.rs]: @l3r8yJ it's already added as puzzle in `cli/src/args.rs`

**Review by l3r8yJ — APPROVED** [review]
@h1alexbel looks good to me!

**Comment by h1alexbel** [comment]
@rultor merge

**Comment by h1alexbel** [comment]
@rultor merge

**Comment by h1alexbel** [comment]
@rultor status

**Comment by h1alexbel** [comment]
@rultor merge

**Comment by h1alexbel** [comment]
@rultor merge

**Comment by h1alexbel** [comment]
@rultor stop

**Comment by h1alexbel** [comment]
@rultor merge

**Pull request merged**